index.js

```diff
@@ -6,15 +6,41 @@ class QuickLRU {
 			throw new TypeError('`maxSize` must be a number greater than 0');
 		}
 
+		if (typeof options.maxAge === 'number' && options.maxAge === 0) {
+			throw new TypeError('`maxAge` must be a number greater than 0');
+		}
+
 		this.maxSize = options.maxSize;
+		this.maxAge = options.maxAge || 0;
 		this.onEviction = options.onEviction;
 		this.cache = new Map();
 		this.oldCache = new Map();
 		this._size = 0;
 	}
 
-	_set(key, value) {
-		this.cache.set(key, value);
+	_hasExpired(key, item) {
+		if (item.expiry <= Date.now()) {
+			if (typeof this.onEviction === 'function') {
+				this.onEviction(key, item.value);
+			}
+
+			this.delete(key);
+			return;
+		}
+
+		return item.value;
+	}
+
+	_set(key, value, expiry) {
+		if (this.maxAge > 0) {
+			this.cache.set(key, {
+				value,
+				expiry: expiry || Date.now() + this.maxAge
```

Comment:
```suggestion
				expiry: expiry || (Date.now() + this.maxAge)
```

to make it more readable